lib/swc/get-position-by-offset.js

```diff
@@ -1,20 +1,24 @@
 'use strict';
 
 module.exports.getPositionByOffset = (offset, source) => {
-    let i = -1;
-    const line = 1;
-    let column = -1;
+    let line = 1;
+    let column = 0;
     
     if (offset > source.length)
         throw Error('end cannot be more then length ' + offset + ', ' + source.length);
     
-    while (i < offset - 1) {
-        ++i;
-        ++column;
+    for (let i = 0; i < offset; i++) {
+        if (source[i] === '\n' && i !== offset - 1) {
+            line++;
+            column = 0;
+        } else {
+            column++;
+        }
     }
     
     return {
         line,
         column,
+        index: offset - 1,
```

Comment:
What this is `index` for?

Reply:
The offset number actually, for some use to re-compute some location if the line and column isn't enough, for example, program insert code.

Reply:
That's said, the start and end or this index must have one for programming insert, for example, vite-plugin-vue-inspector, use this to insert an attribute in tranform time.